### PR TITLE
fix(gatsby): compile themes through default webpack config

### PR DESCRIPTION
**Pull request by grantglidewell**
## Description

Apply default webpack config to gatsby themes @ChristopherBiscardi 

## Related Issues

Addresses:
https://github.com/ChristopherBiscardi/gatsby-theme-examples/issues/10

### CODE REVIEW

**Comment by grantglidewell** [comment]
I dont think the test failure is related to the code changes: 
```
make: Entering directory '/home/circleci/project/node_modules/sharp/build'
  TOUCH Release/obj.target/libvips-cpp.stamp
  CXX(target) Release/obj.target/sharp/src/common.o
../src/common.cc:654:1: error: expected ‘}’ at end of input
 }  // namespace sharp
 ^
```

File: packages/gatsby/src/utils/webpack.config.js [hunk 1]
```diff
@@ -259,6 +259,20 @@ module.exports = async (
       rules.media(),
       rules.miscAssets(),
     ]
+    if (
+      store.getState().config._experimentalThemes &&
```
Comment (ChristopherBiscardi) [inline packages/gatsby/src/utils/webpack.config.js]: We should make this the same as the [component shadowing check](https://github.com/gatsbyjs/gatsby/blob/bc7b0956efa29fcd5c74dabda8a6641fc2cccb19/packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/gatsby-node.js#L9).

```suggestion
      store.getState().themes.themes
```

File: packages/gatsby/src/utils/webpack.config.js [hunk 2]
```diff
@@ -259,6 +259,20 @@ module.exports = async (
...
+      store.getState().config._experimentalThemes.length
```
Comment (ChristopherBiscardi) [inline packages/gatsby/src/utils/webpack.config.js]: and then we can remove this as well.
```suggestion
```

File: packages/gatsby/src/utils/webpack.config.js [hunk 3]
```diff
@@ -259,6 +259,19 @@ module.exports = async (
       rules.media(),
       rules.miscAssets(),
     ]
+    if (
+      store.getState().themes.themes
+    ) {
+      configRules.concat(
```
[the remark this answers is not in the document]
Comment (grantglidewell) [reply packages/gatsby/src/utils/webpack.config.js]: yup, true as concat would not alter the original array

File: packages/gatsby/src/utils/webpack.config.js [hunk 4]
```diff
@@ -259,6 +259,19 @@ module.exports = async (
...
+        store.getState().themes.themes.map(theme => {
+          return {
+            test: /\.jsx?$/,
+            includes: theme.themeDir,
```
[the remark this answers is not in the document]
Comment (grantglidewell) [reply packages/gatsby/src/utils/webpack.config.js]: thanks, good catch

**Comment by wardpeet** [comment]
```
Successfully published:
 - gatsby@2.4.6
```

**Pull request merged**